src/ex3/item-manager-commander.js

```diff
@@ -0,0 +1,61 @@
+import { writeFile, readFileSync } from 'fs';
+
+const DATA_FILE_NAME = "savedData.json";
+
+export class ItemManagerCommander {
+  init() {
+    try {
+      this.items = this.getItemsFromFile();
+    } catch (error) {
+      this.items = [];
+      this.writeItemsToFile();
+    }
+    return this.items;
+  }
+
+  addItem(text) {
+    const itemIndex = this.items.findIndex(item => item.text === text);
+    if (itemIndex > -1) {
+      this.items[itemIndex].isNew = true;
+    } else {
+      this.items.push({text: text, isNew: true });
+    }
+    this.writeItemsToFile();
+    return this.items;
+  }
+
+  markItemAsOld(item){
+    item.isNew = false;
```

Comment:
in general it's a good practice to avoid mutations of data, this helps to eliminate many common bugs. instead, try to create a new copy of your data ex: updatedItem = { ...item, isNew: false }.

Reply:
If I create a new item then I need to delete an existing one and write it again to the list? So it will move to the end of the list. Or I should sort it somehow every time or insert it not at the end of the list but at its original place.

Reply:
as this applies to this.items as well, any change to the array should be done in an immutable way
you can add a few helper functions for adding,updating and removing an item from this list.
example:
removing: `this.items = this.items.filter(oItem => oItem !== item)`
adding: `this.items = { ...this.items, item }`
updating: `this.items = this.item.map(oItem => oItem.id === item.id ? item : oItem)`

Reply:
Ah, thank you, I got it. I will merge the code now as is and add a new branch to fix it.

Reply:
done